src/main/java/com/google/sps/database/DatabaseImpl.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Implementation for DatabaseImpl class
+ *  Current public functions at a glance:
+ *  test()
+ *  addVideo()
+ *  addClause()
+ *  addClauses()
+ *  addMetadata()
+ *  getAllKeywords()
+ *  getTimesForKeyword()
+ *  videoInDb()
+ * 
+ */
+
+package com.google.sps.database;
+
+import com.google.appengine.api.datastore.*;
+import java.util.*;
+
+public class DatabaseImpl implements DatabaseInterface {
+
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final String V_KIND = "video";
+    private static final String M_KIND = "metadata";
+    private static final String C_KIND = "caption";
+    private static final String T_KIND = "timestamps";
+
+    //================================================================================
+    // General Test Function (not used for official testing)
+    //================================================================================
+    
+    public void test() {
+        
+        Map<String, List<Long>> hm = new HashMap<String, List<Long>>();
+
+        String caption = "John Cena";
+        List<Long> timestamps = new ArrayList<>();
+        timestamps.add(42L);
+        timestamps.add(365L);
+        timestamps.add(31415L);
+        hm.put(caption, timestamps);
+
+        String caption2 = "Dwayne 'the Rock' Johnson";
+        List<Long> timestamps2 = new ArrayList<>();
+        timestamps2.add(0L);
+        timestamps2.add(123L);
+        timestamps2.add(404L);
+        hm.put(caption2, timestamps2);
+
+        Entity myEnt = new Entity(V_KIND,"ncbb5B85sd2");
+
+        datastore.put(myEnt);
+
+        List<Entity> le = new ArrayList<>();
+
+        for (Map.Entry<String, List<Long>> me : hm.entrySet()) {
+            Entity tempEnt = new Entity(C_KIND, me.getKey(), myEnt.getKey());
+            tempEnt.setProperty(T_KIND, me.getValue());
+            datastore.put(tempEnt);
+            le.add(tempEnt);
+        }
+        
+        for (Entity e : le) {
+            System.out.println("Parent: " + e.getParent());
+        }
+
+    }
+
+    //================================================================================
+    // Public Interface Functions (see DatabaseInterface.java for official descriptions)
+    //================================================================================    
+
+    // add a video to the database
+    public int addVideo(String videoID, String metadata) {
+        try {
+            Entity vidEnt = new Entity(V_KIND, videoID);
+            datastore.put(vidEnt);
+        } catch (Exception e) {
+            System.out.println("Failed to add video to the database");
+            return -1;
+        }
+
+        try {
+            Entity metaEnt = new Entity(M_KIND, metadata, vidEnt.getKey());
+            datastore.put(metaEnt);
+        } catch (Exception e) {
```

Comment:
Also I would recommend to have an DatabaseException class that contains the error message and error type. It would be easier to identify and figure out the errors.